### ISSUE TITLE
keybinding 'alt' collides with keybinding of atom-core

### ISSUE THREAD

**Issue by colin-kiegel** [atom-community/linter#648]
linter adds this keybinding: **alt**	linter:set-bubble-transparent	Linter	atom-text-editor:not(.mini)

It collides with
**Atom > Settings > Core Settings > Auto Hide Menu Bar**
Automatically hide the menu bar and toggle it by pressing **Alt**. This is only supported on Windows & Linux.

This combination results in not being able to toggle the menu bar within editor windows. This might affect many users of atom, because it is a core keybinding. I suggest to change the default binding for linter to something else.

**Comment by steelbrain** [comment]
What alternative key would you suggest for a peek below the bubble command?

**Comment by colin-kiegel** [comment]
I don't have a preference, because I am quite new to atom (~2 hrs) and just setting it up and getting to know it :-)

**Comment by k2b6s9j** [comment]
On Windows? Or everywhere?

**Comment by steelbrain** [comment]
Everywhere.

**Comment by k2b6s9j** [comment]
Ctrl is free on my OS X install.

**Comment by k2b6s9j** [comment]
Alt was taken up by Nuclide's click to symbol.

**Comment by steelbrain** [comment]
Yeah, same story on linux and windows but not sure, Can someone else confirm that?

**Comment by colin-kiegel** [comment]
I tried to map CTRL on Linux, but that did not work for me for some reason (Ubuntu 15.04)

**Comment by colin-kiegel** [comment]
What about switching to *toggle*-behaviour for transparency? It could then be a more complex keybinding, like CTRL+ALT+[a-Z0-9]

In this case please consider, that keyboard layouts can be quite different around the globe (mostly for special characters) - standard letters should generally be better than special characters. In fact I had to override the other keybinding alt-shift-, because shift-, is interpreted as ; with my current layout and the keybinding did not work, too. But that did not seem as relevant as this one, so I did not file it as a bug. My point is: If you are looking for new keybindings, please try to avoid shift + special-char, because that could lead to unexpected behaviour on different keyboard-layouts.

**Comment by colin-kiegel** [comment]
Another idea:
* default to 'transparent bubbles' when the cursor is moved
* on 'next-error' switch to 'opaque bubbles'
** if the bubble was transparent before (i.e. the cursor had moved) than *stay* at the current position
** else go to the next error

I think that might work, too.

**Comment by steelbrain** [comment]
@colin-kiegel I have just tested mapping ctrl and it works flawlessly.
Here's my keymap.cson file
```coffee
'atom-text-editor:not(.mini)':
  'ctrl': 'linter:set-bubble-transparent'
```

**Comment by colin-kiegel** [comment]
that does *not* work for me (neither left-ctrl or right-ctrl) - I don't know why - but this does
```
'atom-text-editor:not(.mini)':
  'alt-ctrl': 'linter:set-bubble-transparent'
```
but only if alt is pressed *before* ctrl - which is not good either

Anyway - I think mapping it to ctrl is better than mapping it to alt to avoid the collision

**Comment by steelbrain** [comment]
What other packages are you using by the way?

**Comment by colin-kiegel** [comment]
build, cargo-test-runner, keyboard-localization, language-rust, linter, linter-rust, merge-conflicts, minimap, racer, run-command2, rust-api-docs-helper, script, vim-mode

it doesn't help if I deactivate everything except linter, linter-rust and language-rust

**Comment by steelbrain** [comment]
Okay so enable `Key Binding Resolver` from the command pallete. and then press the control key, which commands do you find associated with that key?

**Comment by colin-kiegel** [comment]
There are no key bindings associated with ctrl. Can it be OS-related? (Ubuntu Gnome 15.04?) It doesn't make sense to me.

I think, you could change it and see if other people complain.

**Comment by steelbrain** [comment]
Weird, I am running the same OS Ubuntu GNOME 15.04 with GNOME 3.16 with Atom 0.208.0 and it works flawlessly, for me and other people I know.

**Comment by k2b6s9j** [comment]
Could be a driver thing.

**Comment by steelbrain** [comment]
@k2b6s9j linux doesn't really have drivers, windows does.

**Comment by despairblue** [comment]
FWIW the `alt+shift+.` does not work for on mac or linux (haven't tested windows).

On mac it inputs `˘` and on linux it's interpreted as `alt+>` and thus not working. There are no conflicting key bindings and I was rather surprised by changing it away from `ctrl+k ctrl+n` in the first place. Since that worked flawlessly and now isn't used by anything I have installed.

**Comment by colin-kiegel** [comment]
@despairblue I think this ticket is actually about a different keybinding - please have a look at https://github.com/AtomLinter/Linter/issues/667 :-)

**Comment by despairblue** [comment]
@colin-kiegel true, I simply put it here because you mentioned a [similar issue](https://github.com/AtomLinter/Linter/issues/648#issuecomment-116121551) and the key binding could be addressed by this ticket as a whole instead of opening a ticket for each non working key binding.

**Comment by mcuznz** [comment]
+1 for resolving the collision with Core "Auto Hide Menu Bar"... honestly don't have a positive suggestion for a replacement though.  Even with "Auto Hide Menu Bar" off though, this binding prevents using, for example, Alt+F to access the File Menu.

I haven't tried switching it to Ctrl, but I suspect it would have similar problems, blocking Ctrl-C for example.  Binding a single command to a Chord Key as a default is probably never a good idea.

**Comment by despairblue** [comment]
@mcuznz I don't think that the `alt` binding is preventing you from using `alt-f`. It's rather this one from core:
```coffee
'atom-text-editor':
  'alt-f': 'editor:move-to-end-of-word'
```

Because e.g. `alt-v` is working for me.

**Comment by mcuznz** [comment]
@despairblue I'll disagree, because alt-f to open the File Menu (or alt-e for Edit, or any other menu item) work just fine as soon as I disable Linter - or for that matter, if I override it's "alt" keybinding.

**Comment by despairblue** [comment]
@mcuznz did you unset `alt-f` as well? I'm just saying that if you disable linter, **reload atom** and **open a file** `alt-f` will not work unless you unset  `editor:move-to-end-of-word`, which is a core binding. It's true that the menu bar is not shown by pressing `alt` because of the linter key binding, but the navigation shortcuts like `alt-e` and or `alt-v` should work nevertheless, with the exception of `alt-f` and `alt-h`, because these are taken up by atom itself.

Could you check the keybinding resolver to make sure that it's really linter that is prevent `alt-f` from working?

![screenshot from 2015-06-30 21-45-13](https://cloud.githubusercontent.com/assets/927609/8440278/7fcf377a-1f71-11e5-83c6-2497cb94bb01.png)
![screenshot from 2015-06-30 21-44-58](https://cloud.githubusercontent.com/assets/927609/8440280/7fd2022a-1f71-11e5-8a30-98373bbe3e04.png)
![screenshot from 2015-06-30 21-44-39](https://cloud.githubusercontent.com/assets/927609/8440279/7fcf9274-1f71-11e5-9b4d-91993b49c5e2.png)

**Comment by mcuznz** [comment]
My apologies, I was a bit unclear earlier.

The Chorded input Alt-F (pressed simultaneously) triggers the core "move-to-end-of-word" function as expected, regardless of Linter's enable/disable state.

With Atom's "Auto Hide Menu Bar" option turned on, the Alt key (single press, not chorded) is required to reveal the Menu Bar, wherein after a brief delay I would typically press "F" to access the file menu, "V" for the view menu, etc.

However, with Linter enabled, pressing the Alt key is captured by "linter:set-bubble-transparent" which prevents the hidden Menu Bar from being revealed, thus making all menu items inaccessible by an additional keystroke.

Further, if I add the following to my personal Keymap File, the default "Show Menu" functionality is restored:

```
'atom-text-editor':
  'alt': 'unset!'
```

This solves the problem, but of course effectively disables this feature of Linter :)

**Comment by despairblue** [comment]
This is weird, since it works for me under linux, but I don't use Ubuntu/Unity. Maybe Unity does something weird there, though I don't see the connection to linter in that case :confused:.

In my case the menu bar does not show when pressing `alt`, but it does as soon as I press `v` as well, opening the view menu. That's also why the `alt` shortcut does not annoy me since I can still open the menu bar that way (also the nuclide is also using it, which is annoying, but not linter's fault)

**Comment by mcuznz** [comment]
FWIW, I'm running Ubuntu 14.04 / Gnome Shell 3.10.4, not Unity ;)

**Comment by despairblue** [comment]
good choice imo :smile:

**Comment by samueloph** [comment]
Got the same problem here, I was already reporting this issue to atom (before running it in safe mode) until I saw this here.

I think this should have a great priority, since more people can experience it and lose time trying to figure out what happened.

I understand that may be hard to choose another shortcut, but the package shouldn't conflict with the atom's core. Right?

As a fast workaround, I commented the shortcut.

**Comment by Fedcomp** [comment]
Any progress on this? i would also like to see this fixed. I tryed to override it, but apparently it's not possible.

**Comment by steelbrain** [comment]
@Fedcomp We just haven't found any alternative key to `alt`, but we are totally open to ideas :)

**Comment by Fedcomp** [comment]
I highly suggest to just comment it untill you find appropriate key, because alt menus is much, much more important than some linter:set-bubble-transparent
It's just frustrating to me that i have either to remove the extension (which would totally help me see rust errors) or work with tabs open (thoose dont support themes and use system colors, so i may see dark theme and eye-hurting white menus).
Imo, general keys, such as cmd, ctrl, alt should never been binded as too common. Especially in extensions.

Or is there a setting in the linter to override the keys? i would totally like to have an option, or way to get rid of alt keybinding.

**Comment by colin-kiegel** [comment]
Since a lot of people already 'complained' (see https://github.com/atom/atom/issues/7750) - I suggest to just use anything else. But to do this rather soon. I think 'being quick' on this is very important, because a lot of people are experiencing atom for the first time right now - and this might just hurt the overall experience. We want people to have a good first impression, right?

So let's just pick the first good alternative that's working for everyone:

For example `alt-?`. On US/GB keyboards this should be close to `alt->` and on DE keyboards (german) it's ok too.

I think it's really really important to move away from `alt` quickly.

**Comment by steelbrain** [comment]
Requesting suggestions for alternatives People.

**Comment by AsaAyers** [comment]
I recommend turning off the feature until it can be done right.

**Comment by steelbrain** [comment]
Fixed by #755
